pyccl/tracers.py

```diff
@@ -495,10 +648,17 @@ def __init__(self, cosmo, dndz, has_shear=True, ia_bias=None,
                               fill_value=0)
 
         if has_shear:
-            # Kernel
             kernel_l = get_lensing_kernel(cosmo, dndz)
-            self.add_tracer(cosmo, kernel=kernel_l,
-                            der_bessel=-1, der_angles=2)
+            if (cosmo['sigma_0'] == 0):
+                # GR case
+                self.add_tracer(cosmo, kernel=kernel_l,
+                                der_bessel=-1, der_angles=2)
+            else:
+                # MG case
+                k = np.logspace(-5, -0.7)
```

Comment:
see my response below.

Reply:
You could even embed this code in `_get_MG_transfer_function` and have it return both `mg_transfer` and `k`. That way you don't have to do this every time.

I would even suggest calling `_get_MG_transfer_function` from `_MG_add_tracer`, since that would also simplify the code a lot.